### PR TITLE
DOCSP-43617 Remove unsupported versions

### PR DESCRIPTION
**Pull request by lindseymoore**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-php-library/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43617>
Staging - <https://deploy-preview-163--docs-php-library.netlify.app/compatibility/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

**Review by mongoKart — APPROVED** [review]
lgtm

**Comment by lindseymoore** [comment]
@jmikola After a review of docs, this was only mention of unnecessary mention of 3.6. Let me know what you think, thanks!

**Comment by jmikola** [comment]
> After a review of docs, this was only unnecessary mention of an older version

I found four references to older server versions in the docs:

 * The introductory paragraph in `decimal128.txt` starts with "MongoDB 3.4 introduced support for..." This can be rewritten to just say "The Decimal128 BSON type is a 128-bit decimal-based..."
 * The description of the `flags` option in `MongoDBDatabase-createCollection.txt` includes "MongoDB 3.0 and later". I think that entire paragraph can be removed, as the manual links no longer contain any content on the subject. I also opened [PHPLIB-1545](https://jira.mongodb.org/browse/PHPLIB-1545) and [PHPLIB-1546](https://jira.mongodb.org/browse/PHPLIB-1546) to deprecate and remove the option in PHPLIB 1.21 an 2.0, respectively. It's a relic of the MMAPv1 storage engine.
 * Also in `MongoDBDatabase-createCollection.txt`, you can remove "This option has been deprecated since MongoDB 3.2" from the description of the `autoIndexId` option.
 * `collation.txt` starts with "MongoDB 3.4 introduced", which can be rewritten as "Collations provide a set of rules..." (preserving the manual link if it's still valid). This is comparable to what I suggested above for Decimal128.

**Comment by lindseymoore** [comment]
@jmikola Is it safe to remove `autoIndexId` from `MongoDBDatabase-createCollection.txt`? I don't see `autoIndexId` on the manual page, and it seems it can only be set to to true now anyways. 
Thanks for flagging the above; we can address the tickets to remove the `flag` option once to the engineering tickets are out of code review.

**Comment by jmikola** [comment]
> Is it safe to remove `autoIndexId` from `MongoDBDatabase-createCollection.txt`? I don't see `autoIndexId` on the manual page, and it seems it can only be set to to true now anyways.

Note that `autoIndexId` can be `false` when creating a collection in the non-replicated `local` database. [PHPLIB-1159](https://jira.mongodb.org/browse/PHPLIB-1159) removes the option in PHPLIB 2.0 and there's already a corresponding docs ticket ([DOCSP-44238](https://jira.mongodb.org/browse/DOCSP-44238)), so I don't think it needs to be touched in this PR.

As for the server manual, it _should_ be documented as it's is still technically party of the `create` command API. I opened [DOCSP-43428](https://jira.mongodb.org/browse/DOCSP-43428) some time ago to track that, as it looks like it was inadvertently removed from the manual in recent years.

File: source/data-formats/decimal128.txt [hunk 1]
```diff
@@ -15,11 +15,10 @@ Decimal128
 Overview
 --------
 
-MongoDB 3.4 introduced support for a Decimal128 BSON type,
-which is a 128-bit decimal-based
-floating-point value capable of emulating decimal rounding with exact precision.
-This functionality is intended for applications that handle :manual:`monetary
-data </tutorial/model-monetary-data>`, such as financial and tax computations.
+Decimal128 BSON type is a 128-bit decimal-based floating-point value capable
```
Comment (jmikola) [inline source/data-formats/decimal128.txt:18]: Should this say "The Decimal128 BSON type"? This reads as if "Decimal128 BSON type" is a proper noun.

**Review by jmikola — APPROVED** [review]
One question about adding "the". Regardless of whether that is changed, LGTM and no need to review a second time.

**Pull request merged**